scylla-cdc-replicator/src/replicator_consumer.rs

```diff
@@ -14,6 +14,32 @@ use scylla::Session;
 
 use scylla_cdc::consumer::*;
 
+struct PreparedStatementCache {
+    queries: HashMap<String, PreparedStatement>,
+    query_string_maker: Box<dyn (Fn(&str) -> String) + Sync + Send>,
+}
```

Comment:
I suggest adding a `PreparedStatementCache::new` method so that you don't have to write `queries: HashMap::new()` every time.

Moreover, you can probably also avoid writing Box multiple times by using `impl` and boxing inside the constructor:

```rust
impl PreparedStatementCache {
    pub fn new(query_string_maker: impl (Fn(&str) -> String) + Sync + Send) -> Self {
        /* function contents left as an exercise for the reader */
    }
}
```

Reply:
This definitions gives me  the following error:
```
error[E0310]: the parameter type `impl Fn(&str) -> String + Sync + Send` may not live long enough
  --> scylla-cdc-replicator/src/replicator_consumer.rs:26:33
   |
23 |     pub fn new(query_string_maker: impl (Fn(&str) -> String) + Sync + Send) -> Self {
   |                                    --------------------------------------- help: consider adding an explicit lifetime bound...: `impl Fn(&str) -> String + Sync + Send + 'static`
...
26 |             query_string_maker: Box::new(query_string_maker),
   |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ...so that the type `impl Fn(&str) -> String + Sync + Send` will meet its required lifetime bounds
```
Is it ok to add `'static` lifetime? I'm not sure if it (changing the argument's lifetime) changes anything, since it's only a constraint for the argument.

Reply:
Yes, requiring static is fine here. `'static` just means that the lifetime of the type is not constrained by other lifetimes. For example, it won't accept closures which capture variables from the stack by reference because those variables will be destroyed when the function goes out of scope (so they have non-static lifetime). In your case you are capturing owned Strings by `move` and the closure manages the lifetime of the strings itself, so `'static` will work perfectly fine.